addons/website_sale_collect/models/sale_order.py

```diff
@@ -56,17 +65,35 @@ def _is_in_stock(self, wh_id):
         :return: Whether all storable products are in stock.
         :rtype: bool
         """
-        return not self._get_unavailable_products(wh_id)
+        return not self._get_unavailable_order_lines(wh_id)
 
-    def _get_unavailable_products(self, wh_id):
-        """ Return the products that are not in stock for the given warehouse.
+    def _get_unavailable_order_lines(self, wh_id):
+        """ Return the order lines with unavailable products for the given warehouse.
 
         :param int wh_id: The warehouse in which to check the stock, as a `stock.warehouse` id.
-        :return: The products that are not in stock.
-        :rtype: product.product
+        :return: The order lines with unavailable products.
+        :rtype: sale.order.line
         """
-        out_of_stock_products = self.order_line.filtered(
-            lambda l: l.is_storable
-            and l.product_uom_qty > l.product_id.with_context(warehouse_id=wh_id).free_qty
-        ).product_id
-        return out_of_stock_products
+        unavailable_order_lines = self.env['sale.order.line']
+        for ol in self.order_line:
+            if ol.is_storable:
+                product_free_qty = ol.product_id.with_context(warehouse_id=wh_id).free_qty
+                if ol.product_uom_qty > product_free_qty:
+                    ol.shop_warning = _(
+                        'Only %(new_qty)s available', new_qty=int(max(product_free_qty, 0))
+                    )
+                    unavailable_order_lines |= ol
+        return unavailable_order_lines
+
+    def _verify_updated_quantity(self, order_line, product_id, new_qty, **kwargs):
+        """ Override of `website_sale_stock` to skip the verification when click and collect
+        is activated. The quantity is verified later. """
+        self.ensure_one()
+        product = self.env['product.product'].browse(product_id)
+        if (
+            product.is_storable
+            and not product.allow_out_of_stock_order
+            and self.website_id.in_store_dm_id
+        ):
+            return new_qty, ''
+        return super()._verify_updated_quantity(order_line, product_id, new_qty, **kwargs)
```

Comment:
`_verify_updated_quantity` calls `_get_cart_and_free_qty` in `website_sale_stock` which use the warehouse of the SO. Therefore I don't think this override is necessary.

Reply:
but when the customer has not selected a pickup point yet? it will use the one set on the website and it will not allow not add it.

Reply:
Good point. I didn't think about that use case... I believe it's weird as it is right now but I think it would be worse to prevent adding and item to the cart so we'll improve that later.